frontends/p4/p4-parse.ypp

```diff
@@ -813,6 +827,34 @@ actionRef
                                            $$ = new IR::ActionListElement(@2, $1, mce); }
     ;
 
+entry
+    : optAnnotations keysetExpression ':' actionInvocation
```

Comment:
1. Actions cannot be generic, so you don't need the actionInvocation with `<>`.
2. bison gives bad error messages; we can give a much better error message in the type-checker. Even if the grammar accepts without (), we can enforce this later.
3. whatever this syntax is for actionInvocation, it should accept the same kinds of expressions as the default_action does. They are both accepting actions with all parameters bound. There is no reason that one of them accepts without () and the other one doesn't. 
4. By using initializer we can remove two productions from the grammar. This is not the right metric to optimize, but it's still useful to have a small grammar.

Please note that this is not really an invocation, so the name of the production is unsuitable. It is just binding the arguments; the invocation happens at runtime.